isaric/parsers/isaric-ccpuk.toml

```diff
@@ -0,0 +1,450 @@
+[adtl]
+name = "isaric-ccpuk"
+description = "ISARIC CCPUK study"
+
+[adtl.tables]
+study = { kind = "constant" }
+subject = { kind = "groupBy", groupBy = "subject_id", aggregation = "lastNotNull" }
+visit = { kind = "groupBy", groupBy = "visit_id", aggregation = "lastNotNull" }
+
+[adtl.defs."Y/N/NK".values]
+1 = true
+2 = false
+
+[study]
+id = "isaric-ccpuk"
+date = "2021-01-01"
+country_iso3 = "GBR"
+
+## SUBJECT
+
+[subject]
+study_id = "isaric-ccpuk"
+country_iso3 = "GBR"
+pathogen = "COVID-19"
+
+[visit]
+country_iso3 = "GBR"
+
+[subject.sex_at_birth]
+field = "sex"
+description = "Sex at Birth"
+values = { 1 = "male", 2 = "female", 3 = "non_binary" }
+
+[subject.subject_id]
+field = "\ufeffsubjid"
+sensitive = true
+description = "Participant Identification Number (PIN) specify CPMS Site code (hyphen) four to six digit number patient number e.g. Y0401-0001."
+
+[subject.enrolment_date]
+field = "dsstdat"
+description = "Date of Enrolment"
+
+[subject.admission_date]
+field = "hostdat"
+description = "Admission date at this facility"
+
+[subject.ethnicity]  # TODO: need to add ethnic other field
+description = "Ethnic groups"
+combinedType = "list"
+excludeWhen = "none"
+fields = [
+    { field = "ethnic___1", values = { 1 = "Arab" } },
+    { field = "ethnic___2", values = { 1 = "Black" } },
+    { field = "ethnic___3", values = { 1 = "East_Asian" } },
+    { field = "ethnic___4", values = { 1 = "South_Asian" } },
+    { field = "ethnic___5", values = { 1 = "West_Asian" } },
+    { field = "ethnic___6", values = { 1 = "Latin_American" } },
+    { field = "ethnic___7", values = { 1 = "White" } },
+    { field = "ethnic___8", values = { 1 = "Aboriginal_First_Nations" } },
+]
+
+[subject.works_microbiology_lab]
+field = "labwork_erterm"
+ref = "Y/N/NK"
+description = "Employed in a microbiology laboratory?"
+
+[subject.works_healthcare]
+field = "healthwork_erterm"
+ref = "Y/N/NK"
+description = "Employed as a healthcare worker?"
+
+[subject.age]
+combinedType = "firstNonNull"
+[[subject.age.fields]]
+field = "calc_age"
+unit = "years"
+description = "Calculated age in years"
+
+[[subject.age.fields]]
+field = "age_estimateyears"
+unit = "years"
+description = "Age/Estimated age"
+
+[subject.age.fields.source_unit]
+field = "age_estimateyearsu"
+values = { 1 = "months", 2 = "years" }
+
+[subject.pregnancy]
+field = "pregyn_rptestcd"
+ref = "Y/N/NK"
+description = "Pregnant ?"
+
+[subject.pregnancy_date_of_delivery]
+comment = "considers delivery before and after admission"
+combinedType = "firstNonNull"
+fields = [
```

Comment:
This part looks different from other combinedType entries ... no square brackets [[ ]]. or it doesn't matter?

Reply:
Double square brackets are list of maps (array of tables, https://toml.io/en/v1.0.0#array-of-tables), so these are identical:

```toml
fields = [
  { field = "hello" },
  { field = "world" }
]
```
and
```toml
[[fields]]
field = "hello"

[[fields]]
field = "world"
```